docs/migrations/v8_to_v10.md

```diff
@@ -1,5 +1,113 @@
 # MORYX 8.x to 10.x
 
+## Async life-cycle
+
+### Changes in StateBase
+
+We now provide a full async implementation of the `StateBase`. To keep constistent naming, the `StateBase` was splitted to `SyncStateBase` and `AsyncStateBase`.
+Due to the reduction of unneccessary interfaces, `IState` was removed and `StateBase` will be used in e.g. `IStateContext` now.
+
+The `StateMachine` class was extended by `WithAsync()` and `ForceAsync()`. The `AsyncStateBase` provides async all the way: `NextStateAsync()`, `OnEnterAsync()`, `OnExitAsync`.\
+**Upgrade hint:** Replace `StateBase<TContext>` by `SyncStateBase<TContext>`.
+
+The same convention was applied to `DriverState`. It was renamed to `SyncDriverState` and a new implementation `AsyncDriverState` was added with full async support.\
+**Upgrade hint:** Replace `DriverState<TContext>` by `SyncDriverState<TContext>`.
+
+### Server Module lifecycle refactored to async methods
+
+The lifecycle methods of ServerModules have been migrated from void to async Task to enable modern asynchronous programming with async/await.
+Since there was no native async context for lifecycle methods, asynchronous calls must be synchronized to preserve the existing start/stop behavior.
+The runtime now supports asynchronous initialization, startup, and shutdown processes (e.g., loading entities from a database during startup).
+
+**Changes in ServerModuleBase:**
+
+- `OnInitialize()` -> `OnInitializeAsync()`
+- `OnStart()` -> `OnStartAsync()`
+- `OnStop()` -> `OnStopAsync()`
+
+````cs
+protected override Task OnInitializeAsync()
+{
+    return Task.CompletedTask;
+}
+
+protected override async Task OnStartAsync()
+{
+    var asyncFoo = Container.Resolve<IAsyncFoo>();
+    await asyncFoo.StartAsync();
+}
+
+protected override async Task OnStopAsync()
+{
+    var asyncFoo = Container.Resolve<IAsyncFoo>();
+    await asyncFoo.StopAsync();
+}
+````
+
+**Changes in IModuleManager:**
+
+- `StartModules()` -> `StartModulesAsync()`
+- `StopModules()` -> `StopModulesAsync()`
+- `InitalizeModule(IServerModule module)` -> `InitalizeModuleAsync(IServerModule module)`
+- `StartModule(IServerModule module)` -> `StartModuleAsync(IServerModule module)`
+- `StopModule(IServerModule module)` -> `StopModuleAsync(IServerModule module)`
+- `ReincarnateModule(IServerModule module)` -> `ReincarnateModuleAsync(IServerModule module)`
+
+The `Main` method in `Program.cs` must be updated to a Task-based signature.
+
+````cs
+public static async Task Main(string[] args)
+{
+    AppDomainBuilder.LoadAssemblies();
+
+    [...]
+
+    var moduleManager = host.Services.GetRequiredService<IModuleManager>();
+    await moduleManager.StartModulesAsync();
+
+    await host.RunAsync();
+
+    await moduleManager.StopModulesAsync();
+}
+````
```

Comment:
```suggestion
````
Note: For projects using top-level statements you can use the new async methods without any further actions.
```